src/DynamoApplications/StartupUtils.cs

```diff
@@ -398,45 +399,55 @@ public static List<Exception> CheckAssemblyForVersionMismatches(Assembly assembl
         private static List<Exception> GetVersionMismatchedReferencesInAppDomain(Assembly assembly, String[] assemblyNamesToIgnore)
```

Comment:
@BogdanZavu sounds like it might be cleaner to get the ALC of the input argument assembly. And then only check for conflicts against all assemblies loaded in that ALC ?

@mjkkirschner would MetadataLoadCOntext help in this case ?

Reply:
I think MLC could work, but if I recall correctly, I think it may not force a load of the referenced assemblies so you may need to make that traversal yourself.

it looks like @BogdanZavu is already getting the ALC of the input assembly, so I agree it's a bit confusing why we even look in the other ALCs - is it because we're assuming the worst case, that some other referenced assembly might get loaded in the future into the current ALC?

Reply:
> I think it may not force a load of the referenced assemblies so you may need to make that traversal yourself.
So I think we never want to force load the referenced assemblies.

> is it because we're assuming the worst case, that some other referenced assembly might get loaded in the future into the current ALC?
That is indeed a possibility but actually we can't really tell for sure so I guess there's no point in issuing a warning.
So I think it's better to simplify and only look at the input ALC as @pinzart90 suggested.